### ISSUE TITLE
Famous Google IP 8.8.8.8 failed to parse 

### ISSUE THREAD

**Issue by sensboston** [P3TERX/GeoLite.mmdb#16]
The database is really good and working pretty well, except one small but a strange thing: the well-known, famous Google's IP 8.8.8.8 is failed to parse! Why?

I settled up small php script to use DB, and here is result: https://senssoft.com/geo/geoip.php?ip=8.8.8.8 

Any advice?

**Comment by dgw** [comment]
Looks fine to me?

```json
{"ip":"8.8.8.8","city":"N\/A","region":"N\/A","country":"United States","latitude":37.751,"longitude":-97.822,"asn":15169,"asn_org":"GOOGLE"}
```

**Comment by sensboston** [comment]
@dgw, ```"city":"N\/A","region":"N\/A``` - nothing bothers you?
As an example: https://ipinfo.io/8.8.8.8 looks way better (yes, I know it's commercial site but)

**Comment by dgw** [comment]
No, the city and region data being "N/A" doesn't bother me. 8.8.8.8 is an anycast address; it _isn't_ bound to a specific location.

First of all, you need to be more specific with what _you_ think is wrong here. Seems to "parse" just fine; is your issue with the data/result?

And secondly, this repo doesn't originate the data. If you have problems with the data accuracy, take it up with MaxMind.

**Comment by sensboston** [comment]
@dgw, I don't get your point at all, sorry. If you don't care about city and region, it's your private business; but talking about database consistency, it's a bug.

**Comment by dgw** [comment]
> > 8.8.8.8 is an anycast address
> 
> Nope, you're completely wrong. It's a Google's address.

The site you linked has it tagged as "anycast". It _is_ an [anycast](https://en.wikipedia.org/wiki/Anycast) address. Pretty much all the well-known global DNS resolvers (Quad9, Cloudflare 1.1.1.1, Google DNS, OpenDNS, …) use anycast routing so clients always get connected to the closest datacenter.

**And again: This repo _does not maintain the data_.** MaxMind chose not to associate a location with Google's anycast IP here, not P3TERX. [Go suggest the correction to MaxMind.](https://www.maxmind.com/en/geoip-data-correction-request) It'll get pulled in here if they issue an update containing the change.

**Comment by sensboston** [comment]
What is your point? "If I don't need city and region that means nobody needs 'em"? Are you maintainer of this DB? No! My question does not apply to you at all.

**Comment by dgw** [comment]
I am trying to help you realize that P3TERX (the repo owner) also does not maintain the database itself… What's here is just a convenience, drawing on data from MaxMind.

But all I can do is try to convey accurate information about where you should go to attempt to have the data fixed. I've told you all I can on that front. If you don't believe me, and [the repo's license information](https://github.com/P3TERX/GeoLite.mmdb/blob/main/README.md#license) also doesn't convince you, by all means wait here for a comment with the "Owner" badge. You will not hear from me again in this thread.

**Comment by sensboston** [comment]
@dgw, I'm not an idiot or simpleton; of course I **do understand** , this DB is stolen or ... <type your thoughts>. I just stated a fact of existing bug/lack of information. The maintainer (or anyone else) can forward (or can not) this to MaxMind. What's all, end of story. I don't think what @P3TERX will do something about this: he is from China and (I think so) give a damn about US copyright laws.
I opened an issue just to state a fact of existing bug, nothing more. I really love this stolen DB! It's pretty useful and easy to use. But - if it has a bug - why do you argue "there is NO bug"?!

**Comment by abdullahdevrel** [comment]
I thought I would share an FYI, as I work for IPinfo. We have data that aligns with the majority opinion about good code standards.

When it comes to `anycast` IP addresses, we clearly label them in our API service. For location information, we have to choose at least one location point. Even though anycast IPs fundamentally cannot be located to a single location by our geolocation process ([ping triangulation](https://ipinfo.io/blog/probe-network-how-we-make-sure-our-data-is-accurate/)), we have hints of every possible place the anycast IPs can be located in.

However, sending an array of cities will disrupt the API response standardization. Therefore, we chose the WHOIS record city address or one of the more popular/frequently location backed by our active measurement. The issue with their database is that you will often see they do not have city information in their city database. So, you might as well use a country database at that point. If you do use a country database, then choose us!

We have a [free IP to Country database](https://ipinfo.io/developers/ip-to-country-database) you can use. It is updated daily, provides full accuracy, and, within the context of this repository, is shared with a CC-BY-SA 4.0 license without an EULA. TLDR on the licensing is that we allow commercial distribution *legally*, which only requires an attribution. 

https://ipinfo.io/products/free-ip-database

Let me know if you have any questions. I would be happy to help!

**Comment by sensboston** [comment]
@abdullahdevrel, thanks for clarification. BTW, even it's an anycast address, but you/ipinfo produces a correct output for the well known Google DNS server 8.8.8.8 and with big probability it's located in the Mountain View (what is really good). 

As for me, I'm not an active user of ip geolocation services at all. It was just a "proof of concept" for one old fashioned guy who still thinks what IP geolocation by using whois is good :)

**Comment by abdullahdevrel** [comment]
Thank you @sensboston. I really appreciate the opportunity to talk with you. We do things a bit differently by using active measurements using our 720+ servers strong network. Ping me anytime in [our community](https://community.ipinfo.io/) if you have any questions about IP geolocation and our services.
